File: 6-Data_structures_arrays/reverse_string.py
def reverse_string(input):
    if type(input) != str or len(input) < 2:
        return "hmm that is not good"

    result = []

    for i in range(len(input) - 1, -1, -1):
        result.append(input[i])

    return "".join(result)

    # input = [input[i] for i in range(len(input) - 1, -1, -1)]
    # return "".join(input)


input = "hello my name is Foo"
print(reverse_string(input))

File: 6-Data_structures_arrays/reverse_string.js
const reverseString = str => {
  if (!str || str.length < 2 || typeof str !== 'string') {
    return 'hmm that is not good';
  }

  const result = [];

  for (let i = str.length - 1; i >= 0; i--) {
    result.push(str[i]);
  }

  return result.join('');
};

// const reverseString = str => str.split('').reverse().join('');

str = 'hello my name is Foo';
console.log(reverse(str));
